### PR TITLE
LPS-198530 

### PR DESCRIPTION
**Pull request by MirnaGama**
https://liferay.atlassian.net/browse/LPS-198530

- LPS-198530 Get user old status and avoid updating it if the workflow status is pending

**Reproduction:**
1. Login with admin user
2. Apply Single Approve workflow to User (CP/Applications/Workflow/ProcessBuilder/Configuration/User/Edit/dropdown=SingleApprover/Save)
3. Create a new user 
4. Use Headless API getUserAccount method (v1.0/user-accounts/${userID})
(curl -X 'GET' 'http://localhost:8080/o/headless-admin-user/v1.0/user-accounts/34029' -H 'accept: application/json' -u 'test@liferay.com:test')
5. Via Headless API try to set status to ‘Active'
Use Headless API patchUserAccount method (v1.0/user-accounts/${userID})
(curl -X 'PATCH' 'http://localhost:8080/o/headless-admin-user/v1.0/user-accounts/33949' -H 'accept: application/json' -H 'Content-Type: application/json' -u 'test@liferay.com:test' -d '${responseFromGet}') -> change 'status' to 'Active'

_Actual behavior:_
Status is set to ‘Active’

_Expected behavior:_
Status cannot be changed while user creation under verification

### CODE REVIEW

**Comment by MirnaGama** [comment]
ci:test:sf

**Comment by MirnaGama** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java [hunk 1]
```diff
@@ -498,6 +498,12 @@ public UserAccount patchUserAccount(
 
 		User user = _userService.getUserById(userAccountId);
 
+		if (user.getStatus() == WorkflowConstants.STATUS_PENDING) {
+			throw new BadRequestException(
+				"User " + user.getUserId() +
```
[the remark this answers is not in the document]
Comment (MirnaGama) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java:503]: Done ! https://github.com/liferay-commerce/liferay-portal/commit/a701c914c52363418f5cc6f968412172880f09a7

**Review by smottal — APPROVED** [review]
LGTM! Thx

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:um

**Comment by CrescenzoRegaLF** [comment]
ci:forward:force

**Pull request closed without merging**